File: Informatica MCIC/ejemplos/tipados/Recursividad.java
package tipados;

public class Recursividad {
    public static void main(String[] args) {
        System.out.println(producto(5,3));
        System.out.println(productoIterativo(5, 3));
    }

    public static int producto(int n, int m){
        if(m == 1)
            return n;
        return producto(n, m-1) + n;
    }

    public static int productoIterativo(int n, int m){
        int valor = 0;
        for(int i = 0; i < m; i++)
            valor += n;

        return valor;
    }
}

File: Informatica MCIC/ejemplos/python/recursividad.py
def producto(n, m):
    if m == 1:
        return n
    return n + producto(n, m-1)

def producto_iterativo(n, m):
    valor = 0
    for i in range(m):
        valor += n
    return valor

print(producto(5, 3))
print(producto_iterativo(5, 3))